pkg/flavor/flavor.go

```diff
@@ -15,8 +14,10 @@ import (
 )
 
 type Methods interface {
-	FindProfile(vm *virtv1.VirtualMachine) (*flavorv1alpha1.VirtualMachineFlavorProfile, error)
-	ApplyToVmi(field *k8sfield.Path, profile *flavorv1alpha1.VirtualMachineFlavorProfile, vmiSpec *virtv1.VirtualMachineInstanceSpec) Conflicts
+	FindAndApplyFlavor(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineInstance) (Conflicts, error)
```

Comment:
Why did you combine finding the flavor spec and applying it?
It was simpler to test when they were separate.

Reply:
I have checked your other PR to see how are these new methods used. For example in `vms-admitter.go` in `applyFlavorToVm()` method: First `FindFlavorSpec(vm)` is called, then `FindPreferenceSpec(vm)` and then later `FindAndApplyFlavor(vm, vmi)` which repeats the same calls.

If the apply method is separate, then the find calls would be done only once. Maybe you can add a free-standing helper function to call find followed by apply?